### ISSUE TITLE
Attach xsd files when integrating CPACS tools

### ISSUE THREAD

**Issue by MarAlder** [rcenvironment/rce#23]
In [issue #484](https://github.com/DLR-SL/CPACS/issues/484) we plan to exclude the toolspecifics from CPACS. This is realized by the `xsd:any` element with `processContents=strict` attribute, i.e. the XML processor must obtain the schema for the required namespaces for a proper validation of the XML files. The tool owners must provide this schema file.

The integration of a CPACS tool in RCE provides the possibility to add a documentation file. The same way tool integrators need a possibility to attach the corresponding schema file. This would enable users can run proper validation.

**Comment by ArneBachmannDLR** [comment]
I like the idea of keeping the schema with the integrated tool in RCE, but this is no strict requirement.
If the XSD is specified as an URL inside the XML, any processor/validator could just download and cache the file, when validation becomes necessary (first run?).
Also this assumes only one schema file is necessary to validate any dataset to this integrated tool, doesn't it?

**Comment by MarAlder** [comment]
Of course, both variants are and should be possible. Attaching several schema files would also be practical (this also applies to the documentation). 

From the user's point of view, workflow integrators in particular want quick access to the corresponding schema files. Here it would be extremely laborious to find the Github/Gitlab/SVN repository first (if such exists or is accessible...), to find the corresponding version of the integrated tool and to finally find the corresponding schema file. Therefore I think that the tool owner should get the possibility to attach one or more schema files when integrating his tool into RCE.

**Comment by janflink** [comment]
An issue has been added to our internal issue tracker: https://mantis.sc.dlr.de/view.php?id=17074
